python/lsst/analysis/tools/analysisParts/shapeSizeFractional.py

```diff
@@ -0,0 +1,177 @@
+from __future__ import annotations
+
+import numpy as np
+
+from typing import cast
+from lsst.analysis.tools.vectorActions.selectors import VectorSelector
+from lsst.analysis.tools.vectorActions.vectorActions import LoadVector, MagColumnNanoJansky
+
+from lsst.pex.config import Field
+from lsst.pipe.tasks.configurableActions import ConfigurableActionField, ConfigurableActionStructField
+
+from ..interfaces import (
+    AnalysisAction,
+    KeyedData,
+    KeyedDataAction,
+    Scalar,
+    ScalarAction,
+    VectorAction,
+    KeyedDataSchema,
+    Vector,
+)
+from ..scalarActions import CountAction, MedianAction, SigmaMadAction
+from ..keyedDataActions import AddComputedVector, KeyedScalars
+from ..vectorActions import SnSelector, StellarSelector, FractionalDifference
+from ..vectorActions.calcShapeSize import CalcShapeSize
+
+
+class _AproxMedian(ScalarAction):
+    vectorKey = Field[str](doc="Key for the vector to perform action on", optional=False)
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        return ((self.vectorKey, Vector),)  # type: ignore
+
+    def __call__(self, data: KeyedData, **kwargs) -> Scalar:
+        mask = self.getMask(**kwargs)
+        value = np.sort(data[self.vectorKey.format(**kwargs)][mask])  # type: ignore
+        x = int(len(value) / 10)
+        return np.nanmedian(value[-x:])
```

Comment:
I think this should be 

return np.nanmedian(value[::x])

Reply:
No, in this case it is doing exactly what it should. it sorts the array, finds points that are considered the "tail" and only wants to find the median of those tail values.

This is used as a heuristic to determine approximately what magnitude corresponds to a certain signal to noise.

Reply:
Oh, I was thrown off by the name of the class suggesting a median